### DISCUSSION TITLE
Will the Metadata info of the "event invoked" made available in the Python version of the FunctionInvokedEventArgs class as well?

### DISCUSSION THREAD

**Discussion by rwb-truelime** [microsoft/semantic-kernel#4413] [Q&A] [answered]
Got an inspiring example in the C# cookbook:

https://github.com/microsoft/semantic-kernel/blob/main/dotnet/samples/KernelSyntaxExamples/Example57_KernelHooks.cs

This is exactly what I am looking for:

```
        void MyPostExecutionHandler(object? sender, FunctionInvokedEventArgs e)
        {
            Console.WriteLine($"{e.Function.Name} : Post Execution Handler - Usage: {e.Result.Metadata?["Usage"]?.AsJson()}");
        }
```
I can't seem to find the possibility to get this info (metadata) in the Python version (yet)?

Any suggestions on getting this information out of the Python version of the Kernel with hooks? Or not possible yet?

Thanks!

**Comment by evchaki** [comment]
@rwb-truelime , great question.  Adding @moonbox3  for any insights here.

**Comment by moonbox3** [comment]
Hi @rwb-truelime, thank you for your question. We are targeting further improvements to the function invoking/invoked handlers for v1 (date TBD). You can find a proposal around how this may be handled in the sk-v1-proposal repo both [here](https://github.com/moonbox3/sk-v1-proposal/blob/main/python/src/hooks/hook_base.py) and [here](https://github.com/moonbox3/sk-v1-proposal/blob/main/python/src/kernel.py#L76). Please note that this is preliminary code, and isn't set in stone. There is a lot still influx, and being worked on currently, which may change the initial proposal.

**Comment by rwb-truelime** [comment]
Hi @moonbox3 , thank you for your reply and information. Good to notice further improvements are planned, great! The code doesn't directly imply any signals to be able to trace generated costs by the SK? Apologies if this is actually the case, I can misinterpret things because of a lack of knowledge on the (complete) SK codebase. Thanks

**Reply by moonbox3** [reply, marked as answer]
This particular code that I shared doesn't have that trace information right now, but we're working on improving the response object, which would hold more information. This response object would available via the hooks to allow users to gather that type of trace/metadata as required.

**Reply by rodgermoore** [reply]
Great, thanks!

**Comment by rodgermoore** [comment]
Answered in https://github.com/microsoft/semantic-kernel/discussions/4413#discussioncomment-8023961
